cosmwasm/packages/wasmi-runtime/src/wasm/contract_operations.rs

```diff
@@ -214,7 +215,25 @@ fn start_engine(
 
     // Create a parity-wasm module first, so we can inject gas metering to it
     // (you need a parity-wasm module to use the pwasm-utils crate)
-    let p_modlue = elements::deserialize_buffer(contract).map_err(|_| EnclaveError::InvalidWasm)?;
+    let mut p_modlue: Module =
+        elements::deserialize_buffer(contract).map_err(|_| EnclaveError::InvalidWasm)?;
+
+    let memory_section = p_modlue
+        .memory_section_mut()
+        .ok_or(EnclaveError::CannotInitializeWasmMemory)?;
+
+    if memory_section.entries().len() != 1 {
+        return Err(EnclaveError::CannotInitializeWasmMemory);
+    }
+
+    let memory_entry = memory_section
+        .entries()
+        .first()
+        .ok_or(EnclaveError::CannotInitializeWasmMemory)?;
+
+    let maximum: u32 = 192; // 12 MiB
+    let minimum: u32 = std::cmp::max(maximum, memory_entry.limits().initial());
```

Comment:
@reuvenpo Is this right?

Reply:
perfect :+1: 
I would even add a "contract requests too much initial memory" error if `memory_entry.limits().initial() > maximum`. you can also move `maximum` to a global `const`

Reply:
I think this should be `min` instead of `max` :thinking:

Reply:
you are correct, i did a stupid.